### DISCUSSION TITLE
Disable process/server id in modeline

### DISCUSSION THREAD

**Discussion by sonofjon** [emacs-lsp/lsp-mode#3729] [Q&A] [answered]
How can I disable showing process/server id in the modeline? I've looked at https://emacs-lsp.github.io/lsp-mode/page/settings/modeline/ but can't find anything relevant. Thanks.

![Screen Shot 2022-09-19 at 09 59 24](https://user-images.githubusercontent.com/7573435/190973818-38811c48-49b7-45ae-96d6-d6bad2e4b63d.png)

**Comment by yyoncho** [comment]
There is `diminish` package - https://github.com/emacsmirror/diminish

**Comment by sonofjon** [comment]
Yeah, thanks, but I was hoping there was a custom variable to modify the default mode-line string. I only want to remove the pid, which I have no use of seeing, but keep the rest. Been looking all over the code, to create my own customization, but cannot find where the LSP mode-line is set. Where is it, please?

**Reply by yyoncho** [reply]
Start from here: https://github.com/yyoncho/lsp-mode/blob/master/lsp-mode.el#L2787

**Reply by sonofjon** [reply]
Thanks!

**Comment by sonofjon** [comment, marked as answer]
For anyone interested, here's my solution.

```
;; Custom LSP modeline print function without process id
(defun aj8/lsp--workspace-print (workspace)
  "Visual representation WORKSPACE."
  (let* ((proc (lsp--workspace-cmd-proc workspace))
         (status (lsp--workspace-status workspace))
         (server-id (-> workspace lsp--workspace-client lsp--client-server-id symbol-name)))
    (if (eq 'initialized status)
        (format "%s" server-id)
      (format "%s/%s" server-id status))))

 ;; Don't show process id in modeline
(advice-add #'lsp--workspace-print :override #'aj8/lsp--workspace-print)
```
